datadog_checks_base/datadog_checks/base/utils/containers.py

```diff
@@ -4,19 +4,59 @@
 from six import iteritems
 
 
+class _FreezeKey(object):
+    # "Why does this class exist?"
+    # To freeze and compute hashes or mutable structures, we use sorting as an intermediary step so that the
+    # order of items in the container doesn't change the final hash (a property we could call "commutativity").
+    # When items in the container are all the same type (eg a list of strings), there is no problem to this.
+    # But when the container has items of mixed types, we'd get a `TypeError` on Python 3. (Not on Python 2 though,
+    # since comparisons there default to returning `False`.)
+    # So this class was introduced to support specific comparison cases that we need to support to fulfill the needs
+    # of integrations that use these freeze/hash helpers.
+
+    def __init__(self, value):
+        self.value = value
+
+    def __lt__(self, other):
+        # type: (_FreezeKey) -> bool
+        try:
+            lt = self.value < other.value
+        except TypeError:
+            # We're on Python 3, and values are of differing types.
+            # Some integrations may using freezing on structures that contain `None`, so we want to support it and
+            # use the same behavior than on Python 2, i.e. `None < <anything>` must return `False`...
+            if self.value is None:
+                # `None < x` -> `False`
+                return False
```

Comment:
Python2 behavior is that None is always lower than anything.

```suggestion
            if self.value is None:
                # `None < x` -> `True`
                return True
```

Reply:
Absolutely! Got myself tangled up in comparison orders :joy:

Reply:
Should also handle the case where both values are `None`, py2 returns `False`

Reply:
As it throws an exception (even if `None` and `None` should have the same type 🤷‍♀️ )